### ISSUE TITLE
fsck.fat doesn't seem to work on F32+VFAT volume

### ISSUE THREAD

**Issue by mikrosk** [mikrosk/dosfstools#1]
When invoked as `fsck.dat i:` where `i` is a 40 GB FAT32 volume, it reports:

rwabs_xhdi: access outside partition (drv = I:)

@th-otto does it seem familiar to you? IIRC you ported the patch to latest dosfstools.

**Comment by mikrosk** [comment]
Environment: latest FreeMiNT + HD Driver 12.02

**Comment by th-otto** [comment]
Yes i did quite some time ago. But IIRC i only did some basic tests using freshly created images.

**Comment by mikrosk** [comment]
Interestingly, I have jumped into this issue again. There's definitely some problem with FAT32, this time it was just a 8 GB partition and still not working.

**Comment by mikrosk** [comment]
Same cause as https://github.com/freemint/e2fsprogs/issues/3.

**Comment by mikrosk** [comment]
Fixed by https://github.com/freemint/dosfstools/commit/c40b64434861c82344dcb8c4cb6e1b14baeeeb52.
